src/rate_limiter/src/lib.rs

```diff
@@ -347,6 +358,23 @@ impl RateLimiter {
         })
     }
 
+    // Arm the timer of the rate limiter with `duration` milliseconds.
+    // If `duration` is `None`, the minimum refill duration will be used.
+    fn activate_timer(&mut self, duration: Option<u64>) {
```

Comment:
You could drop the extra complexity of the `duration: Option<u64>` parameter. It involves the caller knowing that `None` maps to a default and `Some` maps to the custom value.

Just take `TimerState` as a param.

```suggestion
    fn activate_timer(&mut self, new_state: TimerState) {
```

Reply:
Done